### PR TITLE
[New Rule] O365 Excessive SSO Logon Errors

### LINKED ISSUE #1150: [New Rule] O365 Excessive SSO Logon Errors
## Description
Detects accounts with high number of Single Sign ON (SSO) logon errors. 

### Target indexes

`filebeat-*, logs-o365*`

### Platforms

Microsoft 365

### Query

```
event.dataset:o365.audit and event.provider:AzureActiveDirectory and event.category:web and
o365.audit.LogonError : "SsoArtifactInvalidOrExpired" 
```

```
[rule.threshold]
field = ["host.id"]
value = 5
```

### **False Positives**
- 
- 

### MITRE

| Tactic | Technique ID | Technique Name | Sub-Technique Name |
| ------ | ------------ | -------------- | ------------------ |
|        |              |                |                    |
|        |              |                |                    |

### References
-

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1150

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+[metadata]
+creation_date = "2021/05/17"
+maturity = "development"
```
Comment (bm11100) [inline rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml:3]:
```suggestion
maturity = "production"
```

File: rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+updated_date = "2021/05/17"
+
+[rule]
+author = ["Elastic"]
```
Comment (bm11100) [inline rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml:7]:
```suggestion
author = ["Elastic", "Austin Songer"]
```

File: rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+description = """
+Identifies accounts with high number of Single Sign ON (SSO) logon errors. Excessive logon errors may indicate 
+attempts to bruteforce of password or single sign on token hijack or reuse.
+"""
+
```
Comment (bm11100) [inline rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml:12]:
```suggestion
false_positives = [
    """
    Automated processes that attempt to authenticate using expired credentials and unbounded retries may lead to false
    positives.
    """,
]
```

Comment (bm11100) [reply rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml:12]: re-lint

File: rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+from = "now-30m"
```
Comment (bm11100) [inline rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml:13]:
```suggestion
from = "now-20m"
```

File: rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+risk_score = 73
+rule_id = "2de10e77-c144-4e69-afb7-344e7127abd0"
+severity = "high"
+tags = ["Elastic", "Cloud", "Microsoft 365", "Continuous Monitoring", "SecOps", "Identity and Access"]
+type = "threshold"
+
+
+query = '''
+event.dataset:o365.audit and event.provider:AzureActiveDirectory and event.category:web and
+o365.audit.LogonError : "SsoArtifactInvalidOrExpired"
```
Comment (bm11100) [inline rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml:29]:
```suggestion
event.dataset:o365.audit and event.provider:AzureActiveDirectory and event.category:web and o365.audit.LogonError:"SsoArtifactInvalidOrExpired" 
```

File: rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+description = """
+Identifies accounts with high number of Single Sign ON (SSO) logon errors. Excessive logon errors may indicate 
+attempts to bruteforce of password or single sign on token hijack or reuse.
+"""
+
+from = "now-30m"
+index = ["filebeat-*", "logs-o365*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "O365 User with Excessive SSO Logon Errors"
```
Comment (bm11100) [inline rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml:17]:
```suggestion
name = "O365 Excessive Single Sign-On Logon Errors"
```

File: rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml [hunk 7]
```diff
@@ -0,0 +1,49 @@
+[metadata]
+creation_date = "2021/05/17"
+maturity = "development"
+updated_date = "2021/05/17"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies accounts with high number of Single Sign ON (SSO) logon errors. Excessive logon errors may indicate 
+attempts to bruteforce of password or single sign on token hijack or reuse.
```
Comment (bm11100) [inline rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml:10]:
```suggestion
Identifies accounts with a high number of single sign-on (SSO) logon errors. Excessive logon errors may indicate an
attempt to brute force a password or single sign-on token.
```

Comment (bm11100) [reply rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml:10]: re-lint

**Comment by bm11100** [comment]
Do you have an example of this triggering?

File: rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml [hunk 8]
```diff
@@ -0,0 +1,52 @@
+[metadata]
+creation_date = "2021/05/17"
+maturity = "production"
+updated_date = "2021/05/17"
+
+[rule]
+author = ["Elastic", "Austin Songer"]
+description = """
+Identifies accounts with a high number of single sign-on (SSO) logon errors. Excessive logon errors may indicate an
+attempt to brute force a password or single sign-on token.
```
Comment (brokensound77) [inline rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml:10]: Does an invalid or expired SSO imply excessive logon errors? Maybe we should re-word this to be more specific to the SSO artifact, which we could still imply may be an attempt to brute force or the result of a phish or some other attack

**Review by brokensound77 — APPROVED** [review]

File: rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    positives.
+    """,
+]
+from = "now-20m"
+index = ["filebeat-*", "logs-o365*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "O365 Excessive Single Sign-On Logon Errors"
+note = """## Config
+The Microsoft 365 Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
```
Comment (brokensound77) [inline rules/microsoft-365/credential_access_user_excessive_sso_logon_errors.toml:24]:
```suggestion

The Microsoft 365 Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
```

**Pull request merged**